src/ModelContextProtocol-SemanticKernel/Extensions/KernelExtensions.cs

```diff
@@ -64,6 +64,137 @@ public static async Task<IReadOnlyList<KernelPlugin>> AddToolsFromClaudeDesktopC
         return registeredPlugins;
     }
 
+    /// <summary>
+    /// Adds Model Content Protocol plugins from the Visual Studio settings file (<c>settings.json</c>) and adds it into the plugin collection.
+    /// </summary>
+    /// <param name="plugins">The plugin collection to which the new plugin should be added.</param>
+    /// <param name="path">Path to the workspace settings file, for user settings use alternative overload</param>
+    /// <param name="loggerFactory">The optional <see cref="ILoggerFactory"/>.</param>
+    /// <param name="cancellationToken">The optional <see cref="CancellationToken"/>.</param>
+    /// <returns>A list of <see cref="KernelPlugin"/> containing the functions provided in plugins.</returns>
+    public static async Task<IReadOnlyList<KernelPlugin>> AddToolsFromVsCodeConfigAsync(
+        this KernelPluginCollection plugins,
+        string? path = null,
+        ILoggerFactory? loggerFactory = null,
+        CancellationToken cancellationToken = default)
+    {
+        if (!File.Exists(path))
```

Comment:
in his case; when the user provides the path, this check is not needed, it can throw exception in the File.OpenRead